qt/aqt/main.py

```diff
@@ -643,6 +640,28 @@ def apply_collection_options(self) -> None:
             Config.Bool.INTERRUPT_AUDIO_WHEN_ANSWERING
         )
 
+    def create_backup_now(self) -> None:
+        """Create a backup immediately, regardless of when the last one was created.
+        Waits until the backup completes. Caller should run this on a background thread."""
+        assert self.col.create_backup(
```

Comment:
You're probably aware as it ties to the remaining decicions you mentioned, but this currently causes Anki to hang.

Reply:
Do you mean that you're seeing the UI become unresponsive when doing a DB check or a full download? They are currently using taskman directly, so for the first part of the operation where the collection lock is held, if the browse screen tries to refresh, it's going to get stuck querying the DB as these two screens aren't using QueryOp().

Or do you mean something else? For the user-initiated create backup case, I've changed the code to do the second half in the background, then monitor for errors. If the user creates a backup and then immediately makes a breaking change before the backup completes, they could potentially be unable to restore the backup if the second half of the process fails, but the window for that to happen would be pretty short.

Reply:
I meant when triggering a manual backup when there have been no changes since the last one, but that's solved with a42c511ceb59c912ea6b476b5487c07637732e96.